ops/hookcmds/_types.py

```diff
@@ -54,14 +54,14 @@ class Address:
     @classmethod
     def _from_dict(cls, d: AddressDict) -> Address:
         return cls(
-            hostname=d['hostname'],
-            value=d['value'],
-            cidr=d['cidr'],
+            hostname=d.get('hostname', ''),
+            value=d.get('value', ''),
+            cidr=d.get('cidr', ''),
         )
 
 
 BindAddressDict = TypedDict(
-    '_BindAddressDict',
+    'BindAddressDict',
     {'mac-address': str, 'interface-name': str, 'addresses': list[dict[str, str]]},
```

Comment:
This seems like it would avert the need to `cast` on line 80.
```suggestion
    {'mac-address': str, 'interface-name': str, 'addresses': list[AddressDict]},
```